EntranceRando.py

```diff
@@ -40,17 +40,21 @@ def __init__(self, rng: random.Random):
     # todo - investigate whether this might leak memory (holds references to Entrances)?
     @staticmethod
     @functools.cache
-    def _is_dead_end(entrance: Entrance):
+    def _is_dead_end(entrance: Entrance, visited_regions=""):
         """
         Checks whether a entrance is an unconditional dead end, that is, no matter what you have,
         it will never lead to new randomizable exits.
         """
-
         # obviously if this is an unpaired exit, then leads to unpaired exits!
         if not entrance.connected_region:
             return False
+        for region in visited_regions.split("::"):
+            if region == entrance.connected_region.name:
+                return True
```

Comment:
This is incorrect, we know that we get to a region we've seen but we don't know for sure as written whether that region has useful exits